src/main/kotlin/io/vlang/configurations/VlangNewToolchainDialog.kt

```diff
@@ -63,14 +67,17 @@ class VlangNewToolchainDialog(private val toolchainFilter: Condition<Path>, proj
             }
         }
 
-        pathToToolchainComboBox.addToolchainsAsync {
+        val disposable = Disposer.newDisposable()
+        mainPanel.registerValidators(disposable)
+
+        val scope = CoroutineScope(SupervisorJob() + Dispatchers.Default)
```

Comment:
I don't think creating your own scope is good. Previously it was also a hack, but best solution would be to have a scope tided to dialog being displayed.
Since I do not have a better solution we can try it out.